### DISCUSSION TITLE
Actions status icons spinning

### DISCUSSION THREAD

**Discussion by UplandJacob** [community/community#144710] [Mobile]
<details>

<summary>

To reproduce:

1. Trigger action with a matrix (has multiple runs)

2. Enter one run's logs while others are still running

3. Wait long enough for others to finish

4. Hit the back arrow: the icons that were the yellow "in progress" icons are now "finished" icons but are still spinning
https://github.com/user-attachments/assets/d44fe057-bece-4146-a8cb-155d791906b0

GitHub for iOS, Version 1.185.1 (147102269)
Device: iPhone 13 (iOS 17.7)

</summary>

</details>

**Comment by BancoCentraI** [comment]
> Para reproduzir:
> 
> 1. Ação de gatilho com uma matriz (tem várias execuções)
> 2. Insira os logs de uma execução enquanto outras ainda estão em execução
> 3. Espere o tempo suficiente para que os outros terminem
> 4. Clique na seta para trás: os ícones que eram os ícones amarelos "em andamento" agora são ícones "concluídos", mas ainda estão girando 
>    https://github.com/user-attachments/assets/d44fe057-bece-4146-a8cb-155d791906b0

Parece que o problema que você está enfrentando envolve uma desatualização de status no ícone de conclusão (que permanece com o "spinner" girando). Esse problema pode ser causado por um atraso na sincronização dos estados da execução, falhas no gerenciamento assíncrono dos logs ou um problema de cache na UI. Dependendo do sistema que você está usando, seria útil revisar a forma como os estados das execuções são atualizados e garantir que todas as execuções estejam sendo tratadas corretamente em paralelo, com atualização de status no momento adequado.

Se você puder fornecer mais detalhes sobre o contexto da plataforma em que isso ocorre, poderei oferecer sugestões mais específicas.

**Reply by UplandJacob** [reply]
I updated the original issue to have the build and such.

Also, this video might give a bit more context. I can reproduce this pretty easily.

https://github.com/user-attachments/assets/bf349c49-d55b-4d0d-bd5c-6fcde8a4b6b0

**Reply by BancoCentraI** [reply]
Possíveis Causas e Soluções
1. Processo de Execução Não Concluído
O ícone amarelo pode indicar que o job está preso em um estado de execução, possivelmente aguardando uma resposta ou concluindo uma tarefa que nunca termina.

Solução: Verifique os logs da execução.
Vá até o GitHub Web (https://github.com/), abra a página do seu PR e clique no workflow do GitHub Actions que está em progresso.
Verifique os logs de cada job. Pode haver um erro ou um estágio que está demorando muito ou travando.
Se algum passo específico estiver travado ou demorando demais, você pode tentar ajustar o fluxo de trabalho (workflow) para identificar ou corrigir a causa do atraso.
2. Timeout de Execução ou Espera por Entrada
Se o seu fluxo de trabalho depende de algum tipo de input interativo, como uma aprovação manual ou uma ação que está esperando uma resposta (por exemplo, aprovação em um job posterior), o GitHub Actions pode ficar "preso" esperando essa interação.

Solução: Verifique se não há etapas que exigem aprovação manual ou que estão esperando uma entrada (como workflow_dispatch ou uma interação manual).
Se possível, remova ou ajuste esses passos para que o workflow seja completamente automatizado.
3. Falha no Job de uma Matriz
Se você estiver utilizando matrizes de execução, o GitHub pode não conseguir finalizar o job até que todas as execuções da matriz sejam concluídas. Mesmo que um job da matriz tenha terminado com sucesso, o status do job geral pode ficar "em progresso" se outros jobs ainda estiverem em andamento ou aguardando.

Solução: Verifique se algum job na matriz está falhando ou demorando mais do que o esperado.
Para isso, você pode adicionar timeouts para os jobs que podem estar demorando ou travando:


jobs:
  your_job:
    timeout-minutes: 30  # Tempo máximo de execução (30 minutos, ajuste conforme necessário)
## Resumo:
Verifique os logs do workflow para identificar se algum passo está travado ou demorando.
Considere adicionar timeouts para evitar que o job fique indefinidamente em progresso.
Se o status no navegador estiver correto, mas no app GitHub para iOS não, pode ser um problema de sincronização do app.
Limpe o cache do GitHub para iOS ou tente reinstalar o aplicativo.
Teste no GitHub Web para ver se o problema está no app para iOS ou no próprio workflow.
Caso o problema persista mesmo após essas verificações, a próxima etapa seria entrar em contato com o suporte do GitHub ou abrir um bug report detalhado, já que isso pode ser um erro específico do sistema do GitHub.

**Reply by UplandJacob** [reply]
The actions finish perfectly fine. The icons themselves update correctly on both GitHub.com and the app. The ONLY thing wrong is that the icons spin in certain cases on the app.

**Reply by UplandJacob** [reply]
It is an unsolved visual bug. I’m not closing it.
